api/handlers.go

```diff
@@ -0,0 +1,288 @@
+package api
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+	"strconv"
+	"time"
+
+	constVal "github.com/sdf1979/go_final_project/const"
+	"github.com/sdf1979/go_final_project/db"
+)
+
+var store *db.Store
+
+func SetStore(storeIn *db.Store) {
+	store = storeIn
+}
+
+func NextDateHandler(writer http.ResponseWriter, request *http.Request) {
+	now := request.FormValue("now")
+	date := request.FormValue("date")
+	repeat := request.FormValue("repeat")
+
+	nowTime, err := time.Parse(constVal.FormatDate, now)
+	if err != nil {
+		http.Error(writer, "Invalid 'now' date format", http.StatusBadRequest)
+		return
+	}
+
+	nextDate, err := NextDate(nowTime, date, repeat)
+	if err != nil {
+		http.Error(writer, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	writer.Write([]byte(nextDate))
+}
+
+func TaskHandler(writer http.ResponseWriter, request *http.Request) {
+	if store == nil {
+		http.Error(writer, "Store not initialized", http.StatusInternalServerError)
+		return
+	}
+
+	switch request.Method {
+	case "POST":
+		taskPost(writer, request)
+	case "GET":
+		taskGet(writer, request)
+	case "PUT":
+		taskPut(writer, request)
+	case "DELETE":
+		taskDelete(writer, request)
+	}
+}
+
+func TasksHandler(writer http.ResponseWriter, request *http.Request) {
+	if store == nil {
+		http.Error(writer, "Store not initialized", http.StatusInternalServerError)
+		return
+	}
+
+	tasksGet(writer, request)
+}
+
+func TaskDoneHandler(writer http.ResponseWriter, request *http.Request) {
+	if store == nil {
+		http.Error(writer, "Store not initialized", http.StatusInternalServerError)
+		return
+	}
+
+	searchId := request.URL.Query().Get("id")
+	if searchId != "" {
```

Comment:
Лучше инвертировать большой if
```suggestion
	if searchId == "" {
		responseWithJson(writer, http.StatusBadRequest, formatErrorForFrontend("task done - empty task id:"))
		return
	}
	...
```